tests/services/scanner/scanner_dao_test.py

```diff
@@ -235,16 +278,248 @@ def test_create_violation_hash_with_inventory_data_not_string(self):
         returned_hash = scanner_dao._create_violation_hash(
             self.test_violation_full_name,
             ['aaa', 'bbb', 'ccc'],
-            self.test_violation_data)                                                  
+            self.test_violation_data)
         self.assertEquals(expected_hash, returned_hash)
 
     def test_create_violation_hash_with_full_name_not_string(self):
         expected_hash = 'f8813c34ab225002fb2c04ee392691b4e37c9a0eee1a08b277c36b7bb0309f9150a88231dbd3f4ec5e908a5a39a8e38515b8e532d509aa3220e71ab4844a0284'
         returned_hash = scanner_dao._create_violation_hash(
             None,
             self.test_inventory_data,
-            self.test_violation_data)                                                  
+            self.test_violation_data)
         self.assertEquals(expected_hash, returned_hash)
 
+    def test_get_latest_scanner_index_id_with_empty_table(self):
+        """The method under test returns `None` if the table is empty."""
+        self.assertIsNone(
+            scanner_dao.get_latest_scanner_index_id(self.session, 'aaa'))
+
+    @mock.patch(
+        'google.cloud.forseti.services.scanner.dao.date_time', autospec=True)
+    def test_get_latest_scanner_index_id(self, mock_date_time):
+        """The method under test returns the newest `ScannerIndex` row."""
+        time1 = datetime.utcnow()
+        time2 = time1 + timedelta(minutes=5)
+        time3 = time1 + timedelta(minutes=7)
+        # ScannerIndex.create() calls get_utc_now_datetime() twice.
+        mock_date_time.get_utc_now_datetime.side_effect = [
+            time1, time1, time2, time2, time3, time3]
+
+        expected_id = time2.strftime(string_formats.TIMESTAMP_MICROS)
+
+        self.session.add(scanner_dao.ScannerIndex.create('aaa'))
+        index2 = scanner_dao.ScannerIndex.create('aaa')
+        index2.scanner_status = IndexState.SUCCESS
+        self.session.add(index2)
+        self.session.add(scanner_dao.ScannerIndex.create('aaa'))
+        self.session.flush()
+        self.assertEquals(
+            expected_id, scanner_dao.get_latest_scanner_index_id(self.session, 'aaa'))
+
+    @mock.patch(
+        'google.cloud.forseti.services.scanner.dao.date_time', autospec=True)
+    def test_get_latest_scanner_index_id_with_specified_state(self, mock_date_time):
```

Comment:
Change `specified` to `failure` as that is very specific to what is being tested here.

Reply:
Done.